### PR TITLE
add slack notification recipe

### PR DESCRIPTION
**Pull request by zentahori**
This is a recipe to notify slack as to when the deployment starts and ends and who does it.
I am not sure which tasks are always called when the deploy starts and ends, no matter what environment the deployment targets.
As far as I know, chefsolo:setup seems to be always called when the deployment starts and chefsolo:cleanup seems to be always called when it ends.

@dmytro 
Please review.

### CODE REVIEW

**Comment by dmytro** [comment]
@zentahori 
For the triggering task, good reference is `airbrake/capistrano` which performs similar function: notification via Airbrake. Please see https://github.com/airbrake/airbrake/blob/master/lib/airbrake/capistrano.rb

File: notify_slack.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ 
+  response
+end
+
+def _slack_channel
+  (fetch(:chef_environment) == "testing") ? "dev_test" : "#dev-deploy"
+end
+
+
+namespace :notify do
```
Comment (dmytro) [inline notify_slack.rb]: Do you think it's better to namespace it similarly to `airbrake/capistrano`?
```
namespace :slack do
  task :deploy do
  end
end
```

Comment (zentahori) [inline notify_slack.rb]: The namespace :slack is better idea.
I would like to distinguish deployment start from its finish. So I need one more nest, such as slack:deploy:start and  slack:deploy:finish.

**Comment by zentahori** [comment]
I changed this a little bit.
Please check it.

**Comment by zentahori** [comment]
I modified the recipe to add the whole channel notification for slack as a result of today's discussion.
@dmytro 
Please review.

**Comment by dmytro** [comment]
:+1:

**Comment by zentahori** [comment]
@dmytro 
I don't have permission to merge this PR.
Please merge this.

**Comment by dmytro** [comment]
Oh, right. Merging.

**Comment by zentahori** [comment]
Thank you!

**Pull request merged**